app/src/main/java/com/geeksville/mesh/ui/MapFragment.kt

```diff
@@ -1,680 +1,316 @@
-            binding = MapViewBinding.inflate(inflater, container, false)
-            binding.root
-        } else {
-            mapNotAllowedBinding = MapNotAllowedBinding.inflate(inflater, container, false)
-            mapNotAllowedBinding.root
-        }
-    }
-
-    var mapView: MapView? = null
-
-
-    private fun removeOfflineRegions() {
-        // Remove the tile region with the tile region ID.
-        // Note this will not remove the downloaded tile packs, instead, it will just mark the tileset
-        // not a part of a tile region. The tiles still exists as a predictive cache in TileStore.
-        tileStore.removeTileRegion(TILE_REGION_ID)
-
-        // Set the disk quota to zero, so that tile regions are fully evicted
-        // when removed. The TileStore is also used when `ResourceOptions.isLoadTilePacksFromNetwork`
-        // is `true`, and also by the Navigation SDK.
-        // This removes the tiles that do not belong to any tile regions.
-        tileStore.setOption(TileStoreOptions.DISK_QUOTA, Value(0))
-
-        // Remove the style pack with the style url.
-        // Note this will not remove the downloaded style pack, instead, it will just mark the resources
-        // not a part of the existing style pack. The resources still exists as disk cache.
-
-        if (userStyleURI != null) {
-            offlineManager.removeStylePack(userStyleURI!!)
-            mapView?.getMapboxMap()?.loadStyleUri(loadMapStyleFromPref())
-        } else {
-            offlineManager.removeStylePack(mapView?.getMapboxMap()?.getStyle()?.styleURI.toString())
-            mapView?.getMapboxMap()?.loadStyleUri(loadMapStyleFromPref())
-        }
-        MapboxMap.clearData(resourceOptions) {
-            it.error?.let { error ->
-                debug(error)
+                points.add(GeoPoint(it.latitude, it.longitude))
+                controller.animateTo(points[0], nodeZoomLevel, defaultZoomSpeed)
             }
         }
-        updateStylePackDownloadProgress(0, 0)
-        updateTileRegionDownloadProgress(0, 0)
     }
 
-    /**
-     * Mapbox native code can crash painfully if you ever call a mapbox view function while the view is not actively being show
-     */
-    private val isViewVisible: Boolean
-        get() = mapView?.isVisible == true
-
-    override fun onViewCreated(viewIn: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(viewIn, savedInstanceState)
-
-        // We might not have a real mapview if running with analytics
-        if ((requireContext().applicationContext as GeeksvilleApplication).isAnalyticsAllowed) {
-//            binding.fabStyleToggle.setOnClickListener {
-//                //TODO: Setup Style menu for satellite view, street view, & outdoor view
-//            }
-            binding.downloadRegion.setOnClickListener {
-                // Display menu for download region
-                this.downloadRegionDialogFragment()
+    private fun setESRITileSource() {
```

Comment:
I would like to move this out to its own class where we can add extra tile sources/layers more easily in the future.